pkg/apis/externaldns/types.go

```diff
@@ -51,6 +54,11 @@ func NewConfig() *Config {
 
 // ParseFlags adds and parses flags from command line
 func (cfg *Config) ParseFlags(args []string) error {
+	defaultDryRun, err := strconv.ParseBool(os.Getenv("DRY_RUN"))
+	if err != nil {
+		defaultDryRun = true
```

Comment:
it is probably safe in this case, but in general should probably fail if conversion fails. Or use another flag lib :D